File: backend/app/schemas/obituary.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class ObituaryCreate(BaseModel):
    name: str
    birth_date: str  # "YYYY-MM-DD"
    death_date: str  # "YYYY-MM-DD"
    is_public: bool = False
    image: Optional[str] = None  # Base64 encoded image or will be file upload
class ObituaryResponse(BaseModel):
    id: str
    user_id: str
    name: str
    birth_date: str
    death_date: str
    obituary_text: str
    image_url: Optional[str]
    audio_url: Optional[str]
    is_public: bool
    created_at: datetime

    class Config:
        from_attributes = True
class ObituaryListResponse(BaseModel):
    obituaries: list[ObituaryResponse]
    total: int

File: frontend/types/obituary.ts
export interface Obituary {
  id: string;
  user_id: string;
  name: string;
  birth_date: string;
  death_date: string;
  obituary_text: string;
  image_url?: string | null;
  audio_url?: string | null;
  is_public: boolean;
  created_at: string;
}

export interface ObituaryListResponse {
  obituaries: Obituary[];
  total: number;
}
